### ISSUE TITLE
Breaking change in `MeanAveragePrecision` by removing `average` param

### ISSUE THREAD

**Issue by Advitya17** [Lightning-AI/torchmetrics#2004]
## 🐛 Bug

We're using the `MeanAveragePrecision` method in Microsoft's [Responsible AI Toolbox](https://github.com/microsoft/responsible-ai-toolbox/blob/main/responsibleai_vision/responsibleai_vision/rai_vision_insights/rai_vision_insights.py#L1110) repo for calculating Object Detection metrics, and realized that a breaking update in torchmetrics seems to remove the `average` attribute.

What would now be a way to specify a 'macro' or 'micro' averaging method for calculating the Object Detection metrics, and is that soon to be addressed in a next release?

### To Reproduce

Steps to reproduce the behavior...

<details>
  <summary>Code sample</summary>

</details>

### Environment

- TorchMetrics version (and how you installed TM, e.g. `conda`, `pip`, build from source):
- Python & PyTorch Version (e.g., 1.0):
- Any other relevant information such as OS (e.g., Linux):

### Additional context

The `average` attribute has suddenly been removed from https://torchmetrics.readthedocs.io/en/stable/detection/mean_average_precision.html, though still exists for other image scenarios - https://torchmetrics.readthedocs.io/en/stable/classification/average_precision.html

**Comment by SkafteNicki** [comment]
Hi @Advitya17, thanks for raising this issue.
I am a bit confused because I am pretty sure that `MeanAveragePrecision` have never had a `average` argument, but I may be wrong.
What version of torchmetrics have you been using?

**Comment by Advitya17** [comment]
torchmetrics==1.0.1, we had specified it in our code quite a few months back and were able to use torchmetrics until the breaking change happened.

**Comment by SkafteNicki** [comment]
Then I do still not understand, because in v1.0.1 of torchmetrics `MeanAveragePrecision` did not have a average argument:
https://github.com/Lightning-AI/torchmetrics/blob/035e433ce416d82e415d16ba87f8c12cbb74fdef/src/torchmetrics/detection/mean_ap.py#L241-L250

**Comment by Advitya17** [comment]
Yes if it can be included in a future release, that'll be very helpful, thank you very much!

**Comment by Borda** [comment]
> Then I do still not understand, because in v1.0.1 of torchmetrics `MeanAveragePrecision` did not have a average argument:

it was taken by the `**kwargs` so it has zero effect but it did not crash...
